packages/format-library/src/text-color/test/index.native.js

```diff
@@ -0,0 +1,170 @@
+		const {
+			getByA11yLabel,
+			getByTestId,
+			getByA11yHint,
+		} = await initializeEditor();
+
+		// Wait for the editor placeholder
+		const paragraphPlaceholder = await waitFor( () =>
+			getByA11yLabel( 'Add paragraph block' )
+		);
+		expect( paragraphPlaceholder ).toBeDefined();
+		fireEvent.press( paragraphPlaceholder );
+
+		// Wait for the block to be created
+		const paragraphBlock = await waitFor( () =>
+			getByA11yLabel( /Paragraph Block\. Row 1/ )
+		);
+		expect( paragraphBlock ).toBeDefined();
+
+		// Look for the highlight text color button
+		const textColorButton = await waitFor( () =>
+			getByA11yLabel( 'Text color' )
+		);
+		expect( textColorButton ).toBeDefined();
+		fireEvent.press( textColorButton );
+
+		// Wait for Inline color modal to be visible
+		const inlineTextColorModal = getByTestId( 'inline-text-color' );
+		await waitFor( () => inlineTextColorModal.props.isVisible );
+
+		// Look for the pink color button
+		const pinkColorButton = await waitFor( () =>
+			getByA11yHint( COLOR_PINK )
+		);
+		expect( pinkColorButton ).toBeDefined();
+		fireEvent.press( pinkColorButton );
+
+		// Dismiss the inline color modal
+		fireEvent( inlineTextColorModal, 'backdropPress' );
+
+		expect( getEditorHtml() ).toMatchSnapshot();
+	} );
+
+	it( 'allows toggling the highlight color feature to selected text', async () => {
+		const {
+			getByA11yLabel,
+			getByTestId,
+			getByPlaceholderText,
+			getByA11yHint,
+		} = await initializeEditor();
+		const text = 'Hello this is a test';
+
+		// Wait for the editor placeholder
+		const paragraphPlaceholder = await waitFor( () =>
+			getByA11yLabel( 'Add paragraph block' )
+		);
+		expect( paragraphPlaceholder ).toBeDefined();
+		fireEvent.press( paragraphPlaceholder );
+
+		// Wait for the block to be created
+		const paragraphBlock = await waitFor( () =>
+			getByA11yLabel( /Paragraph Block\. Row 1/ )
+		);
+		expect( paragraphBlock ).toBeDefined();
+
+		// Update TextInput value
+		const paragraphText = getByPlaceholderText( 'Start writing…' );
+		fireEvent( paragraphText, 'onSelectionChange', 0, text.length, text, {
+			nativeEvent: {
+				eventCount: 1,
+				target: undefined,
+				text,
+			},
+		} );
+
+		// Look for the highlight text color button
+		const textColorButton = await waitFor( () =>
+			getByA11yLabel( 'Text color' )
+		);
+		expect( textColorButton ).toBeDefined();
+		fireEvent.press( textColorButton );
+
+		// Wait for Inline color modal to be visible
+		const inlineTextColorModal = getByTestId( 'inline-text-color' );
+		await waitFor( () => inlineTextColorModal.props.isVisible );
+
+		// Look for the pink color button
+		const pinkColorButton = await waitFor( () =>
+			getByA11yHint( COLOR_PINK )
+		);
+		expect( pinkColorButton ).toBeDefined();
+		fireEvent.press( pinkColorButton );
+
+		// Dismiss the inline color modal
+		fireEvent( inlineTextColorModal, 'backdropPress' );
```

Comment:
Similar to the comment on the previous test, looks like we won't really need to dismiss the modal.